jdi-light/src/main/java/com/epam/jdi/light/settings/WebSettings.java

```diff
@@ -2,231 +2,279 @@
+            webSettings = new WebSettings();
+        }
+        return webSettings;
+    }
+
+    public String getDomain() {
+        if (getJDISettings().DRIVER.domain != null)
+            return getJDISettings().DRIVER.domain;
         init();
         return "No Domain Found. Use test.properties or JDISettings.DRIVER.domain";
     }
-    public static void setDomain(String domain) {
-        DRIVER.domain = domain;
+
+    public void setDomain(String domain) {
+        getJDISettings().DRIVER.domain = domain;
     }
-    public static VisualCheckAction VISUAL_ACTION_STRATEGY = VisualCheckAction.NONE;
-    public static VisualCheckPage VISUAL_PAGE_STRATEGY = VisualCheckPage.NONE;
-    public static boolean STRICT_SEARCH = true;
-    public static boolean hasDomain() {
+
+    public boolean hasDomain() {
         init();
-        return DRIVER.domain != null && DRIVER.domain.contains("://");
+        return jdiSettings.DRIVER.domain != null && jdiSettings.DRIVER.domain.contains("://");
     }
-    public static String TEST_GROUP = "";
-    // TODO multi properties example
-    public static Safe<String> TEST_NAME = new Safe<>((String) null);
-    public static String useDriver(JFunc<WebDriver> driver) {
-        return WebDriverFactory.useDriver(driver);
+
+    public String useDriver(JFunc<WebDriver> driver) {
+        return getWebDriverFactory().useDriver(driver);
     }
-    public static String useDriver(String driverName) {
-        return WebDriverFactory.useDriver(driverName);
+
+    public String useDriver(String driverName) {
+        return getWebDriverFactory().useDriver(driverName);
     }
-    public static String useDriver(DriverTypes driverType) {
-        return WebDriverFactory.useDriver(driverType);
+
+    public String useDriver(DriverTypes driverType) {
+        return getWebDriverFactory().useDriver(driverType);
     }
-    public static String printSmartLocators(IBaseElement el) {
+
+    public String printSmartLocators(IBaseElement el) {
         try {
-            return "smart: " + format(ELEMENT.smartTemplate, ELEMENT.smartName.value.execute(el.getName()));
+            return "smart: " + format(getJDISettings().ELEMENT.smartTemplate, getJDISettings().ELEMENT.smartName.value.execute(el.getName()));
         } catch (Exception ex) {
-            return format("Can't define smart locator(%s, %s)", ELEMENT.smartTemplate, ELEMENT.smartName.key);
+            return format("Can't define smart locator(%s, %s)", getJDISettings().ELEMENT.smartTemplate, getJDISettings().ELEMENT.smartName.key);
         }
     }
-    public static JFunc1<IBaseElement, WebElement> SMART_SEARCH = el -> {
-        if (ELEMENT.useSmartSearch == FALSE ||
-            ELEMENT.useSmartSearch == ONLY_UI && el.base().locator == null ||
-            ELEMENT.useSmartSearch == UI_AND_ELEMENTS && el.base().locator == null && isInterface(el.getClass(), PageObject.class))
+
+    public JFunc1<IBaseElement, WebElement> SMART_SEARCH = el -> {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5482899)